### PR TITLE
feat(core): add formatDuration utility with comprehensive tests

### PR DESCRIPTION
**Pull request by prod-blip**
## Summary
- Add `formatDuration` function to `packages/core/src/utils.ts` that converts milliseconds to human-readable duration strings (e.g., `"1h 2m 3s"`)
- Add comprehensive unit tests covering edge cases: zero, negative, NaN, Infinity, partial seconds, large durations
- This is the inverse of the existing `parseDuration` function in `lifecycle-manager.ts`

## Test plan
- [x] Run `pnpm test --filter @composio/ao-core` - 26 utils tests pass
- [x] Run `pnpm typecheck` - passes
- [x] Run lint on modified files - passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/core/src/utils.ts [hunk 1]
```diff
@@ -133,6 +133,34 @@ export async function readLastJsonlEntry(
...
+  const minutes = Math.floor((totalSeconds % 3600) / 60);
+  const seconds = totalSeconds % 60;
+
+  const parts: string[] = [];
+  if (hours > 0) parts.push(`${hours}h`);
+  if (minutes > 0) parts.push(`${minutes}m`);
+  if (seconds > 0 || parts.length === 0) parts.push(`${seconds}s`);
+
+  return parts.join(" ");
+}
```
[the remark this answers is not in the document]
Comment (prod-blip) [reply packages/core/src/utils.ts:162]: Fixed in 5c0168cc. Added `formatDuration` to the barrel file export in `index.ts` to follow the established pattern.
